### PR TITLE
Add docs for streaming

### PR DESCRIPTION
**Pull request by collindutter**
----
:books: Documentation preview :books:: https://griptape--112.org.readthedocs.build/

### CODE REVIEW

**Review by andrewfrench — APPROVED** [review]
Good examples!

**Comment by collindutter** [comment]
Examples use new event structure from [here](https://github.com/griptape-ai/griptape/pull/375), blocked until that is merged.

**Pull request merged**
